From 872863f266bc5902c94e673c76fa5a079db7772b Mon Sep 17 00:00:00 2001
From: Larry Woodman <lwoodman@redhat.com>
Date: Mon, 19 Jan 2015 14:28:53 -0500
Subject: [mm] Revert: vmstat: create fold_diff

Message-id: <1421677734-17441-4-git-send-email-lwoodman@redhat.com>
Patchwork-id: 103150
O-Subject: [RHEL7.1 PATCH 3/4] Revert "[mm] vmstat: create fold_diff"
Bugzilla: 1179654
RH-Acked-by: Rafael Aquini <aquini@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

This reverts commit 318b151a21bd5558801e6af939e8a7f0606026ec.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/mm/vmstat.c b/mm/vmstat.c
index 3f30142..750482b 100644
--- a/mm/vmstat.c
+++ b/mm/vmstat.c
@@ -414,15 +414,6 @@ void dec_zone_page_state(struct page *page, enum zone_stat_item item)
 EXPORT_SYMBOL(dec_zone_page_state);
 #endif
 
-static inline void fold_diff(int *diff)
-{
- int i;
-
- for (i = 0; i < NR_VM_ZONE_STAT_ITEMS; i++)
-  if (diff[i])
-   atomic_long_add(diff[i], &vm_stat[i]);
-}
-
 /*
  * Update the zone counters for the current cpu.
  *
@@ -492,7 +483,10 @@ static void refresh_cpu_vm_stats(int cpu)
    drain_zone_pages(zone, &p->pcp);
 #endif
  }
- fold_diff(global_diff);
+
+ for (i = 0; i < NR_VM_ZONE_STAT_ITEMS; i++)
+  if (global_diff[i])
+   atomic_long_add(global_diff[i], &vm_stat[i]);
 }
 
 /*
@@ -522,7 +516,9 @@ void cpu_vm_stats_fold(int cpu)
    }
  }
 
- fold_diff(global_diff);
+ for (i = 0; i < NR_VM_ZONE_STAT_ITEMS; i++)
+  if (global_diff[i])
+   atomic_long_add(global_diff[i], &vm_stat[i]);
 }
 
 /*
-- 
1.7.1